x-pack/plugins/cases/public/components/create/form_context.tsx

```diff
@@ -106,9 +106,11 @@ export const FormContext: React.FC<Props> = ({
 
       for (const [key, value] of Object.entries(customFields)) {
         const configCustomField = customFieldsConfiguration.find((item) => item.key === key);
-        const fieldValue = isEmpty(value) ? null : [value];
 
         if (configCustomField) {
+          const fieldValue =
+            configCustomField.type === CustomFieldTypes.TEXT && isEmpty(value) ? null : [value];
```

Comment:
There was a bug where you switch the toggle field as on during create case and it still remained `off` in case view page.
Because it was set to null here.

Reply:
Instead of checking specific for a custom field type we can do `isEmpty(value) && value !== false ? null : [value]`

Reply:
This check will still have issue when `toggle` field value is `true` when creating a case, I updated it to  `isEmpty(value) && typeof value !== 'boolean' ? null : [value];`

When we have more custom fields in future, we can create some util function to handle different types.